File: src/main/kotlin/top/potmot/controller/EntityController.kt
package top.potmot.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import top.potmot.model.dto.GenEntityConfigInput
import top.potmot.model.dto.GenEntityPropertiesInput
import top.potmot.model.dto.GenEntityPropertiesView
import top.potmot.model.query.EntityQuery
import top.potmot.service.EntityService

@RestController
@RequestMapping("/entity")
class EntityController (
    @Autowired val entityService: EntityService
) {
    @PostMapping("/map")
    fun map(@RequestBody tableIds: List<Long>): List<GenEntityPropertiesView> {
        return entityService.mapEntities(tableIds)
    }

    @GetMapping("/sync")
    fun sync(@RequestBody tableIds: List<Long>): List<GenEntityPropertiesView> {
        return entityService.syncEntities(tableIds)
    }

    @PutMapping("/save")
    fun save(@RequestBody entities: List<GenEntityPropertiesInput>): List<GenEntityPropertiesView> {
        return entityService.saveEntities(entities)
    }

    @PutMapping("/config")
    fun config(@RequestBody entity: GenEntityConfigInput): GenEntityPropertiesView {
        return entityService.configEntity(entity)
    }

    @PostMapping("/query")
    fun query(@RequestBody query: EntityQuery): List<GenEntityPropertiesView> {
        return entityService.queryEntities(query, GenEntityPropertiesView::class)
    }

    @DeleteMapping("/{ids}")
    fun delete(@PathVariable ids: List<Long>): Int {
        return entityService.deleteEntities(ids)
    }
}

File: table-graph/src/__generated/services/EntityController.ts
import type { Executor } from '../';
import type { EntityQuery, GenEntityConfigInput, GenEntityPropertiesInput, GenEntityPropertiesView, GenTableColumnsInput, Optional } from '../model/static';

export class EntityController {
    
    constructor(private executor: Executor) {}
    
    async config(options: EntityControllerOptions['config']): Promise<
        Optional<GenEntityPropertiesView>
    > {
        let _uri = '/entity/config';
        return (await this.executor({uri: _uri, method: 'PUT', body: options.body})) as Optional<GenEntityPropertiesView>
    }
    
    async delete(options: EntityControllerOptions['delete']): Promise<number> {
        let _uri = '/entity/';
        return (await this.executor({uri: _uri, method: 'DELETE', body: options.body})) as number
    }
    
    async map(options: EntityControllerOptions['map']): Promise<
        GenEntityPropertiesView
    > {
        let _uri = '/entity/map';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as GenEntityPropertiesView
    }
    
    async query(options: EntityControllerOptions['query']): Promise<
        ReadonlyArray<GenEntityPropertiesView>
    > {
        let _uri = '/entity/query';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as ReadonlyArray<GenEntityPropertiesView>
    }
    
    async save(options: EntityControllerOptions['save']): Promise<
        ReadonlyArray<Optional<GenEntityPropertiesView>>
    > {
        let _uri = '/entity/save';
        return (await this.executor({uri: _uri, method: 'PUT', body: options.body})) as ReadonlyArray<Optional<GenEntityPropertiesView>>
    }
    
    async sync(options: EntityControllerOptions['sync']): Promise<
        ReadonlyArray<GenEntityPropertiesView>
    > {
        let _uri = '/entity/sync/';
        _uri += encodeURIComponent(options.tableId);
        return (await this.executor({uri: _uri, method: 'GET'})) as ReadonlyArray<GenEntityPropertiesView>
    }
}

export type EntityControllerOptions = {
    'config': {readonly body: GenEntityConfigInput},
    'delete': {readonly body: ReadonlyArray<number>},
    'map': {readonly body: GenTableColumnsInput},
    'query': {readonly body: EntityQuery},
    'save': {readonly body: ReadonlyArray<GenEntityPropertiesInput>},
    'sync': {readonly tableId: number}
}